component/otelcol/connector/grafanacloud/grafanacloud.go

```diff
@@ -0,0 +1,94 @@
+// Package grafanacloud provides an otelcol.connector.grafanacloud component.
+package grafanacloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/connector"
+	"github.com/grafana/river"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.connector.grafanacloud",
```

Comment:
FWIW: our naming convention for components expects multiple words to be separated by underscores; for this to align with the convention this should be `otelcol.connector.grafana_cloud`

Reply:
thanks @rfratto. It looks like with otelcol components, there may be some inconsistency. If we wrap contrib components then do we want to match the names? contrib omits the underscores.

Reply:
We've noticed the inconsistency for sure :) It's something we're looking into resolving. Matching the upstream components exactly isn't necessary; Flow components are already named differently (with the namespaces) and we can use documentation to inform people about what the wrapped component is called.

Reply:
Sounds good! once the final name is decided, I'll make sure we follow the conventions here.